scoretracker/tracker.py

```diff
@@ -1,95 +1,164 @@
-                else:
-                    for sco in database[person].keys():
-                        rT['text'] += '    '+sco+': '+database[person][sco]+'\n'
-            else:
-                for per, scores in database.items():
-                    rT['text'] += person + ':\n'
-                    for sco in scores:
-                        rT['text'] += '    '+sco+': '+database[per][sco]+'\n'
-    except Exception:
-        invalid_request()
-        return
-    requests.post('https://api.groupme.com/v3/bots/post', json.dumps(rT))
+    """
+    Sends to the chat the information about:
+    * All (user, score) combinations
+    * All scores of a given user
+    * A given user's score in a specific topic
+
+    @ed> /score show ron
+    @scorebot> ron:
+                   disappointment: 420
+                   understanding-the-rules: -69
+    """
+    response = {'bot_id': BOT_ID}
+    response['text'] = ''
+
+    database = get_db()
+
+    if not database:
+        # Backup did not exist or was empty
+        response['text'] = 'There are no scores yet!'
+
+    elif person is None:
+        response['text'] = '\n'.join(
+            '{person}:\n{scores}'.format(person=person, scores=get_scores(database, person))
+            for person in database
+        )
+
+    elif person not in database:
+        response['text'] = 'User is either unrecognized or has not points'
+
+    elif score is None:
+        response['text'] = get_scores(database, person, multi=False)
+
+    else:
+        response['text'] = '{person}:\n    {score}: {points}'.format(
+            person=person, score=score, points=database[person][score]
+        )
+
+    requests.post(POST_URL, json.dumps(response))
 
 
 def invalid_request():
-    rT = {}
-    rT['bot_id'] = BOT_ID
-    rT['text'] = 'That was not a valid scorebot command'
-    requests.post('https://api.groupme.com/v3/bots/post', json.dumps(rT))
+    """
+    Respond to the chat informing the user that they entered an invalid
+    command for the bot.
+    """
+    response = {
+        'bot_id': BOT_ID,
+        'text': 'That was not a valid scorebot command'
+    }
+    requests.post(POST_URL, json.dumps(response))
 
 
-# /score <person> <score> <+/-> <#>
-# /score show
-# /score show <person> [score]
 def recv_msg():
+    """
+    Handles the CRUD operations represented by the following commands:
+
+    Increment/Decrement a score:
+        /score <person> <score> <+/-> <#>
+
+    Show all user's scores:
+        /score show
+
+    Show a specific person's score(s):
+        /score show <person> [score]
+    """
     message = connexion.request.json['text']
     message = message.split(' ')
     if message[0] == '/score':
         if len(message) == 5:
             print("Change scores")
-            score = message[2]
-            person = message[1]
-            number = int(message[4])
-            if message[3] == '-':
+            _, person, score, plus_minus, number = message
```

Comment:
Didn't know you could do this -- very cool

Reply:
I'm quite partial to unpacking assignment myself. I tend to find it clearer in small cases then a long series of index lookups